integtest/src/main/java/org/corfudb/universe/cluster/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,233 @@
+package org.corfudb.universe.cluster.docker;
+
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.LogStream;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.corfudb.universe.cluster.ClusterException;
+import org.corfudb.universe.node.CorfuServer;
+import org.corfudb.universe.node.NodeException;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static lombok.Builder.Default;
+import static org.corfudb.universe.cluster.Cluster.ClusterParams;
+
+/**
+ * Implements a docker instance representing a Corfu CorfuServer.
+ */
+@Slf4j
+@Builder
+public class CorfuServerDockerized implements CorfuServer {
+
+    @Getter
+    private final ServerParams params;
+    private final DockerClient docker;
+    private final ClusterParams clusterParams;
+    @Default
+    private final Optional<String> containerId = Optional.empty();
+
+    /**
+     * Deploys a Corfu server / docker container
+     */
+    @Override
+    public CorfuServerDockerized deploy() throws NodeException {
+        log.info("Deploying the Corfu server. {}", params.getGenericName());
+
+        if(containerId.isPresent()){
+            throw new IllegalStateException("Corfu server already started. Parameters: " + params);
+        }
+
+        String id = deployContainer();
+
+        CorfuServerDockerized server = CorfuServerDockerized.builder()
+                .clusterParams(clusterParams)
+                .params(params)
+                .docker(docker)
+                .containerId(Optional.of(id))
+                .build();
+
+        server.addShutdownHook();
+        return server;
+    }
+
+    /**
+     * This method attempts to gracefully stops the Corfu server. After timeout, it will kill the Corfu server.
+     *
+     * @param timeout a duration after which the stop will kill the server
+     * @throws NodeException this exception will be thrown if the server can not be stopped.
+     */
+    @Override
+    public void stop(Duration timeout) throws NodeException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.universe.node.NodeException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)